packages/services/src/storage.service.ts

```diff
@@ -1,58 +1,43 @@
+import { IStorageService, StorageServiceConfig } from './types/storage.js';
+import { getStorageUrlByPath } from './utils/helpers.js';
 import { publicAPI } from './utils/httpService.js';
 
-export type Json = {
-  [key: string]: any;
-};
 
-export interface IUploadedJson {
-  filename: string;
-  bucket: string;
-  json: Json;
-  storageAPIUrl: string;
-}
-/**
- * Uploads a json file to S3 storage.
- *
- * @param filename
- * @param json
- * @param bucket
- * @param storageAPIUrl
- * @param typeBucket
- * @returns link to the uploaded json file
- *
- * @example
- * const json = {
- *  "name": "John",
- *  "age": 30,
- * }
- * const url = await uploadJson('test', json, BucketName.PublicVC, 'https://storage.com', 'bucket-verifiable-credentials');
- * // Returns: https://storage.com/test.json
- */
-export const uploadJson = async ({ filename, bucket, json, storageAPIUrl }: IUploadedJson): Promise<string> => {
+export const storageService: IStorageService = async (config) => {
+  const { url, params, options = { method: 'POST', headers: [] } } = config;
   try {
-    if (!storageAPIUrl) throw new Error('storageAPIUrl is not defined');
-
-    if (!bucket) throw new Error('bucket is not defined');
-
-    const file = new File([JSON.stringify(json)], `${filename}`, {
-      type: 'application/json',
-    });
-
-    const presignedUrlParams = {
-      bucket: bucket,
-      key: file.name,
-      fileType: file.type,
-    };
-    const { presignedUrl } = await publicAPI.post<{ presignedUrl: string }>(
-      `${storageAPIUrl}/presigned-url`,
-      presignedUrlParams,
-    );
+    let result;
+    switch (options.method) {
+      case 'PUT':
+        result = await publicAPI.put<string>(url, params, options);
+        break;
+      case 'POST':
+        result = await publicAPI.post<string>(url, params, options);
+        break;
+      default:
+        throw new Error(`Unsupported method`);
+    }
 
-    publicAPI.setContentTypeHeader(file.type);
-    await publicAPI.put(presignedUrl.toString(), file);
-
-    return `https://${bucket}.s3.ap-southeast-2.amazonaws.com/${file.name}`;
+    const getResultByPath = getStorageUrlByPath(result, params.resultPath);
+    return getResultByPath;
   } catch (error: any) {
-    throw new Error(error.message ?? 'Error uploading json');
+    console.error(error);
+    throw new Error(error.message);
   }
```

Comment:
I think this code block doesn't make sense. It can reduce the information of the error when the error is thrown to the caller of this function. So, I think this function doesn't need to use a try-catch block unless you want to customize the error message or add special logic in the catch block (e.g., debugging).

Reply:
Yes, after checking the function that calls `storageService` function, I think we'll refactor it later. Some events such as `Transformation Events`, it do not have try/catch. We cannot handle these errors without `try/catch` in `storageService`.